src/components/passwordstrengthmeter.js

```diff
@@ -0,0 +1,64 @@
+import React, { useEffect, useRef } from "react"
+import zxcvbn from "zxcvbn"
+import "../styles/passwordstrengthmeter.css"
+
+function PasswordStrengthMeter({ password, onStrengthUpdate }) {
+
+  const score = useRef(0)
+
+  // check if password meets strength requirement and then tell parent
+  const onStrengthUpdateHandler = (input) => {
+
+   // deal with the case that an error in email or username resests the input to 0
+    if (input !== 0) {
+      return onStrengthUpdate(createPasswordLabel(input))
+    }
+  }
+
+  // convert the password strength to an informative label
+  const createPasswordLabel = (input) => {
+    switch (input) {
+      case 0:
+        return "Weak"
+      case 1:
+        return "Weak"
+      case 2:
+        return "Fair"
+      case 3:
+        return "Good"
+      case 4:
+        return "Strong"
+      default:
+        return "Weak"
+    }
+  }
+
+  useEffect(() => {
+    onStrengthUpdateHandler(score.current)
+    if (password === "") {
+      score.current = 0
+    } else {
+      score.current = zxcvbn(password).score
+    }
+  }, [password])
```

Comment:
Codacy found an issue: [React Hook useEffect has a missing dependency: 'onStrengthUpdateHandler'. Either include it or remove the dependency array.](https://app.codacy.com/gh/treetoplearning/heartwood/pullRequest?prid=5919976)